overlord/devicestate/devicestate.go

```diff
@@ -202,11 +203,28 @@ func checkGadgetOrKernel(st *state.State, snapInfo, curInfo *snap.Info, _ snap.C
 	return nil
 }
 
+func checkGadgetValid(st *state.State, snapInfo, _ *snap.Info, snapf snap.Container, flags snapstate.Flags, deviceCtx snapstate.DeviceContext) error {
+	if snapInfo.GetType() != snap.TypeGadget {
+		// not a gadget, nothing to do
```

Comment:
This semantic was slightly surprising at first given the promise of the function name, but I see it's part of the general check snap callback chain, so it has to be that way.